### PR TITLE
220 add an options menu

### PR DESCRIPTION
**Pull request by 31Nathan**
We now have an option menu
It's horrible for now, but when we will have sprites and font it will be better
A player isn't considered as AFK in menus

### CODE REVIEW

File: Client/ButtonAction.cpp [hunk 1]
```diff
@@ -6,16 +6,33 @@
 */
 
 #include "ButtonAction.hpp"
-#include <mutex>
+#include <csignal>
```
Comment (LucasTesnier) [inline Client/ButtonAction.cpp:9]: Why did you remove the mutex include ?

File: Client/ClientRoom.cpp [hunk 2]
```diff
@@ -166,8 +168,8 @@ void ClientRoom::_initSpritesForEntities()
         sf::Vector2f(0, 0), sf::Vector2f(1920, 1080));
     spritesList.addTexture(GraphicsTextureResource::BACKGROUND_LAYER_1, "assets/Backgrounds/middle.png",
         sf::Vector2f(0, 0), sf::Vector2f(1920, 1080));
-    spritesList.addTexture(GraphicsTextureResource::EXIT_BUTTON, "assets/EpiSprite/r-typesheet11.gif",
-        sf::Vector2f(34, 0), sf::Vector2f(34, 34));
+    spritesList.addTexture(GraphicsTextureResource::BUTTON, "assets/EpiSprite/r-typesheet11.gif", sf::Vector2f(34, 0),
```
Comment (LucasTesnier) [inline Client/ClientRoom.cpp:171]: 34 by 0 ? Are you sure ?

File: libs/R-TypeLogic/EntityManipulation/ButtonManipulation/Components/DisplayState.hpp [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+#ifndef DISPLAYSTATE_HPP_
+#define DISPLAYSTATE_HPP_
+
+#include "Component/Component.hpp"
+#include "R-TypeLogic/EntityManipulation/ButtonManipulation/SharedResources/MenuStates.hpp"
+
+namespace ecs
+{
+
```
Comment (LucasTesnier) [inline libs/R-TypeLogic/EntityManipulation/ButtonManipulation/Components/DisplayState.hpp:16]: useless empty line

File: libs/R-TypeLogic/Global/Systems/NoAfkInMenuSystem.cpp [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        std::lock_guard(*entityPtr.get());
+        GameClock &gameClock = world.getResource<GameClock>();
+        AfkFrequency &freq = entityPtr->getComponent<AfkFrequency>();
+
+        double delta = freq.frequency.count() - gameClock.getElapsedTime();
+
+        if (delta <= 5.0) {
+            Velocity &vel = entityPtr.get()->getComponent<Velocity>();
+
+            vel.modified = true;
```
Comment (LucasTesnier) [inline libs/R-TypeLogic/Global/Systems/NoAfkInMenuSystem.cpp:33]: Smart move :eyes:

**Review by LucasTesnier — CHANGES_REQUESTED** [review]
Some minor questions, otherwise LGTM

File: libs/GraphicECS/SFML/Systems/InputManagement.cpp [hunk 5]
```diff
@@ -110,12 +111,14 @@ namespace graphicECS::SFML::Systems
             std::queue<std::pair<ActionQueueComponent::inputAction_e, float>> &actions =
                 entityPtr->getComponent<ActionQueueComponent>().actions;
             while (actions.size() > 0) {
-                if (actions.front().first == ActionQueueComponent::MOVEY)
-                    movePlayerY(world, actions.front().second);
-                if (actions.front().first == ActionQueueComponent::MOVEX)
-                    movePlayerX(world, actions.front().second);
-                if (actions.front().first == ActionQueueComponent::SHOOT)
-                    shootAction(world, actions.front().second);
+                if (world.getResource<MenuStates>().currentState == MenuStates::IN_GAME) {
```
Comment (ZQUEMA) [inline libs/GraphicECS/SFML/Systems/InputManagement.cpp:114]: You have to lock the Resource.

File: libs/GraphicECS/SFML/Systems/InputManagement.cpp [hunk 6]
```diff
@@ -192,10 +195,12 @@ namespace graphicECS::SFML::Systems
             std::lock_guard(*entityPtr.get());
             Position &pos = entityPtr.get()->getComponent<Position>();
             Size &size = entityPtr.get()->getComponent<Size>();
+            DisplayState &state = entityPtr.get()->getComponent<DisplayState>();
+
             bool sameWidth = pos.y <= mousePos.y && mousePos.y <= pos.y + size.y;
             bool sameHeigth = pos.x <= mousePos.x && mousePos.x <= pos.x + size.x;
 
-            if (sameHeigth && sameWidth) {
+            if (sameHeigth && sameWidth && state.displayState == world.getResource<MenuStates>().currentState) {
```
Comment (ZQUEMA) [inline libs/GraphicECS/SFML/Systems/InputManagement.cpp:203]: You have to lock the Resource.

File: libs/R-TypeLogic/EntityManipulation/ButtonManipulation/Components/DisplayState.hpp [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    class DisplayState : public Component{
```
Comment (ZQUEMA) [inline libs/R-TypeLogic/EntityManipulation/ButtonManipulation/Components/DisplayState.hpp:16]: does it respect the clang format ?

File: Client/ClientRoom.cpp (hunk 2 above)
Comment (31Nathan) [reply Client/ClientRoom.cpp:171]: It's the position not the size

File: Client/ButtonAction.cpp (hunk 1 above)
Comment (31Nathan) [reply Client/ButtonAction.cpp:9]: Because it was useless

File: libs/R-TypeLogic/Global/Systems/NoAfkInMenuSystem.cpp [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+#include "R-TypeLogic/Global/SharedResources/GameClock.hpp"
+#include "R-TypeLogic/Server/Components/AfkFrequencyComponent.hpp"
+
+using namespace ecs;
+
+void NoAfkInMenu::run(World &world)
+{
+    std::vector<std::shared_ptr<Entity>> joined = world.joinEntities<AfkFrequency>();
+
+    if (world.getResource<MenuStates>().currentState == MenuStates::IN_GAME)
```
Comment (ZQUEMA) [inline libs/R-TypeLogic/Global/Systems/NoAfkInMenuSystem.cpp:20]: Lock the Resource

File: libs/R-TypeLogic/Global/Systems/NoAfkInMenuSystem.cpp [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+        return;
+
+    auto updatePlayersInMenu = [&world](std::shared_ptr<Entity> entityPtr) {
+        std::lock_guard(*entityPtr.get());
+        GameClock &gameClock = world.getResource<GameClock>();
```
Comment (ZQUEMA) [inline libs/R-TypeLogic/Global/Systems/NoAfkInMenuSystem.cpp:25]: Lock the Resource

**Review by ZQUEMA — CHANGES_REQUESTED** [review]
Lock every entities and Resources

File: libs/R-TypeLogic/EntityManipulation/ButtonManipulation/Components/DisplayState.hpp (hunk 7 above)
Comment (31Nathan) [reply libs/R-TypeLogic/EntityManipulation/ButtonManipulation/Components/DisplayState.hpp:16]: Does it pass the CI ?

**Review by ZQUEMA — DISMISSED** [review]

File: libs/R-TypeLogic/EntityManipulation/ButtonManipulation/Components/DisplayState.hpp (hunk 7 above)
Comment (ZQUEMA) [reply libs/R-TypeLogic/EntityManipulation/ButtonManipulation/Components/DisplayState.hpp:16]: It seems but doesn't need a space right after the `publi Component` ?

File: libs/GraphicECS/SFML/Systems/InputManagement.cpp (hunk 5 above)
Comment (ZQUEMA) [reply libs/GraphicECS/SFML/Systems/InputManagement.cpp]: Yes ! even in comparison

**Review by ZQUEMA — APPROVED** [review]

File: libs/GraphicECS/SFML/Components/GraphicsTextComponent.hpp [hunk 10]
```diff
@@ -26,7 +26,7 @@ namespace graphicECS::SFML::Components
         /// @param x The x position of the text on the window. Default 10.
         /// @param y The y position of the text on the window. Default 10.
         /// @param size The character size of the text. Default 16.
-        GraphicsTextComponent(const std::string newText = "", const std::size_t x = 10, const std::size_t y = 10,
+        GraphicsTextComponent(const sf::Font &newFont, const std::string newText = "", const std::size_t x = 10, const std::size_t y = 10,
```
Comment (osvegn) [inline libs/GraphicECS/SFML/Components/GraphicsTextComponent.hpp:29]: Documentation hasn't been updated

File: libs/R-TypeLogic/EntityManipulation/ButtonManipulation/Components/DisplayState.hpp [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+
+#ifndef DISPLAYSTATE_HPP_
+#define DISPLAYSTATE_HPP_
+
+#include "Component/Component.hpp"
+#include "R-TypeLogic/EntityManipulation/ButtonManipulation/SharedResources/MenuStates.hpp"
+
+namespace ecs
+{
+    class DisplayState : public Component {
```
Comment (osvegn) [inline libs/R-TypeLogic/EntityManipulation/ButtonManipulation/Components/DisplayState.hpp:16]: Missing class documentation

**Review by osvegn — CHANGES_REQUESTED** [review]
Missing documentation, otherwise it's good

**Review by osvegn — APPROVED** [review]
Look good

**Review by ZQUEMA — APPROVED** [review]

**Review by TomDUVAL-MAHE — APPROVED** [review]

**Review by LucasTesnier — APPROVED** [review]
LGTM
